contracts/src/tokens/drop/DropFacetBase.sol

```diff
@@ -90,57 +94,123 @@ abstract contract DropFacetBase is IDropFacetBase {
     }
   }
 
+  function _verifyPenaltyBps(
+    ClaimCondition storage condition,
+    Claim calldata claim,
+    uint16 expectedPenaltyBps
+  ) internal view returns (uint256 amount) {
+    if (condition.penaltyBps != expectedPenaltyBps) {
+      CustomRevert.revertWith(DropFacet__UnexpectedPenaltyBps.selector);
+    }
+
+    amount = claim.quantity;
+
+    uint16 penaltyBps = condition.penaltyBps;
+    if (penaltyBps > 0) {
+      unchecked {
+        uint256 penaltyAmount = BasisPoints.calculate(
+          claim.quantity,
+          penaltyBps
+        );
+        amount = claim.quantity - penaltyAmount;
+      }
+    }
+
+    return amount;
+  }
+
+  function _addClaimCondition(
+    DropStorage.Layout storage ds,
+    ClaimCondition calldata newCondition
+  ) internal {
+    (uint48 existingStartId, uint48 existingCount) = (
+      ds.conditionStartId,
+      ds.conditionCount
+    );
+    uint48 newConditionId = existingStartId + existingCount;
+
+    // Check timestamp order
+    if (existingCount > 0) {
+      ClaimCondition storage lastCondition = ds.conditionById[
+        newConditionId - 1
+      ];
```

Comment:
```suggestion
      ClaimCondition storage lastCondition;
      unchecked {
          lastCondition = ds.conditionById[newConditionId - 1];
      }
```
This can be unchecked.